File: Exemplos M10/App_IntentDigitalHUB_XamarinAndroid/Elgin_IntentDigitalHUB_XamarinAndroid/IntentServices/Termica/ImprimeXMLNFCe.cs
namespace Xamarin_Android_Intent_Digital_Hub.IntentServices.Termica
{
    class ImprimeXMLNFCe : TermicaCommand
    {
        readonly string dados;
        readonly int indexcsc;
        readonly string csc;
        readonly int param;

        public ImprimeXMLNFCe(string dados, int indexcsc, string csc, int param) : base("ImprimeXMLNFCe")
        {
            this.dados = dados;
            this.indexcsc = indexcsc;
            this.csc = csc;
            this.param = param;
        }

        protected override string FunctionParameters()
        {
            return "\"dados\"" + ":" + "\"" + dados + "\"" + "," +
                    "\"indexcsc\"" + ":" + indexcsc + "," +
                    "\"csc\"" + ":" + "\"" + csc + "\"" + "," +
                    "\"param\"" + ":" + param;
        }
    }
}

File: Exemplos M10/App_IntentDigitalHUB_ReactNative/Elgin_IntentDigitalHUB_ReactNative-font/src/intentDigitalHubService/TERMICA/ImprimeXMLNFCe.ts
import {TermicaCommand} from './TermicaCommand';

export class ImprimeXMLNFCe extends TermicaCommand {
  readonly dados: string;
  readonly indexcsc: number;
  readonly csc: string;
  readonly param: number;

  constructor(dados: string, indexcsc: number, csc: string, param: number) {
    super('ImprimeXMLNFCe');
    this.dados = dados;
    this.indexcsc = indexcsc;
    this.csc = csc;
    this.param = param;
  }

  functionParameters(): object {
    return {
      dados: this.dados,
      indexcsc: this.indexcsc,
      csc: this.csc,
      param: this.param,
    };
  }
}
